### PR TITLE
fix: update project and group membership settings

### PR DESCRIPTION
**Pull request by leafty**
On project settings:
* Do not show the member controls when the user does not have sufficient permissions
    Screenshot from 2024-10-22 14-57-53

On group settings:
* Use similar controls for group members as what is used for project members
    Screenshot from 2024-10-22 15-01-51

/deploy renku=release-0.59.0

### CODE REVIEW

**Comment by leafty** [comment]
(WIP description)

On project settings:
* Do not show the member controls when the user does not have sufficient permissions
    Screenshot from 2024-10-22 14-57-53

On group settings:
* Use similar controls for members as what is used for project members

File: client/src/features/groupsV2/fields/EditGroupMemberModal.tsx [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+          },
+        ],
+      });
+    },
+    [groupSlug, patchGroupMembers]
+  );
+
+  return (
+    <>
+      <ModalBody className="pb-0">
```
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/fields/EditGroupMemberModal.tsx:121]: The bottom padding is removed here and added in the nested div. I'd suggest removing both

```suggestion
      <ModalBody>
```

File: client/src/features/groupsV2/fields/EditGroupMemberModal.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        <Form noValidate onSubmit={handleSubmit(onSubmit)}>
+          {error && <RtkOrNotebooksError error={error} />}
+          <div
+            className={cx("align-items-baseline", "d-flex", "flex-row", "mb-3")}
```
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/fields/EditGroupMemberModal.tsx:125]:
```suggestion
            className={cx("align-items-baseline", "d-flex", "flex-row")}
```

File: client/src/features/groupsV2/fields/AddGroupMemberModal.tsx
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/fields/AddGroupMemberModal.tsx:1]: Nitpick: can you remove the `mb-3` at line 155? Just to keep the visualization consistent with the Edit modal where you adjusted bottom spacing

File: client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      userId: member.id,
+    });
+  }, [deleteGroupMember, groupSlug, member.id]);
+
+  return (
+    <>
+      <ModalBody>
+        {error && <RtkOrNotebooksError error={error} />}
+        <div
+          className={cx("align-items-baseline", "d-flex", "flex-row", "mb-3")}
```
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx:104]:
```suggestion
          className={cx("align-items-baseline", "d-flex", "flex-row")}
```

File: client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        >
+          <Label>
+            Remove <ProjectMemberDisplay member={member} nameInBold={true} />{" "}
+            from group?
+          </Label>
```
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx:109]: Why a `Label` here? It also adds unnecessary spacing.
I'd suggest using a `<p className="mb-0">` 

```suggestion
          <p className="mb-0">
            Remove <ProjectMemberDisplay member={member} nameInBold={true} />{" "}
            from group?
          </p>
```

File: client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        </div>
+      </ModalBody>
+      <ModalFooter>
+        <Button className="btn-outline-rk-green" onClick={toggle}>
```
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx:113]:
```suggestion
        <Button color="outline-primary" onClick={toggle}>
```

File: client/src/features/groupsV2/settings/GroupSettingsMembers.tsx [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+            <GroupMemberItem
+              key={member.id}
+              group={group}
+              index={idx}
+              member={member}
+              members={data}
+              onDelete={onDelete}
+            />
+          ))}
+        </ListGroup> */}
```
Comment (lorenzo-cavazzi) [inline client/src/features/groupsV2/settings/GroupSettingsMembers.tsx:129]: Leftover?

```suggestion
```

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
The code looks good overall! Just a couple of visual details to adjust (see inline).

One nitpick detail: when changing the access level, on the group page the Change button is disabled until you pick a different access level. On the project page, it's always enabled (see screenshot). It's just a detail, but perhaps you can update the project page too
Screenshot_20241023_171902

File: client/src/features/groupsV2/fields/EditGroupMemberModal.tsx (hunk 1 above)
Comment (leafty) [reply client/src/features/groupsV2/fields/EditGroupMemberModal.tsx:121]: Question about these style adjustments: should these be updated on the `Project` components too? The modals where adapted from the `Project` case.

File: client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx (hunk 4 above)
Comment (leafty) [reply client/src/features/groupsV2/fields/RemoveGroupMemberModal.tsx:109]: This was taken from the `Project` version.

File: client/src/features/groupsV2/fields/EditGroupMemberModal.tsx (hunk 1 above)
Comment (lorenzo-cavazzi) [reply client/src/features/groupsV2/fields/EditGroupMemberModal.tsx]: Keeping them consistent would be great if you don't mind touching that code too.

**Comment by lorenzo-cavazzi** [comment]
> Question about these style adjustments: should these be updated on the `Project` components too? The modals where adapted from the `Project` case.

It's great if you want to fix those too, to keep them all consistent, although I don't want to force you to inflate the PR with changes on other components.

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm :rocket:

**Pull request merged**